ugs-core/src/com/willwinder/universalgcodesender/GrblController.java

```diff
@@ -640,8 +665,11 @@ private void handleStatusString(final String string) {
             if (attemptsRemaining > 0 && lastLocation != null) {
                 attemptsRemaining--;
                 // If the machine goes into idle, we no longer need to cancel.
-                if (this.controllerStatus.getState().equals("Idle")) {
+                if (this.controllerStatus.getState().equals("Idle") || this.controllerStatus.getState().equalsIgnoreCase("Check")) {
                     isCanceling = false;
+
+                    // Make sure the GUI gets updated
+                    this.dispatchStateChange(getControlState());
```

Comment:
Do you remember which edge case this happens in?

Reply:
I think this was when pressing the cancelSend-button when paused it will normally change from states "Hold" to "Idle" and this will automatically dispatch a state change event. But when using "Check" we are simulating the "Hold/Pause" and no state change event will be triggered and we need to do that manually.